Buildings/Applications/DHC/CentralPlants/Cooling/Controls/ChilledWaterPumpSpeed.mo

```diff
@@ -0,0 +1,91 @@
+within Buildings.Applications.DHC.CentralPlants.Cooling.Controls;
+model ChilledWaterPumpSpeed
+  "Controller for variable speed chilled water pumps"
+  extends Modelica.Blocks.Icons.Block;
+
+  parameter Integer numPum(min=1, max=2)=2 "Number of chilled water pumps, maximum is 2";
+  parameter Modelica.SIunits.PressureDifference dpSetPoi(displayUnit="Pa")
+    "Pressure difference setpoint";
+  parameter Modelica.SIunits.Time tWai "Waiting time";
+  parameter Modelica.SIunits.MassFlowRate m_flow_nominal
+    "Nominal mass flow rate of single chilled water pump";
+  parameter Real minSpe(unit="1",min=0,max=1) = 0.05
+    "Minimum speed ratio required by chilled water pumps";
+  parameter Modelica.Blocks.Types.SimpleController controllerType=
+    Modelica.Blocks.Types.SimpleController.PID
+    "Type of pump speed controller";
+  parameter Real k(unit="1", min=0) = 1 "Gain of controller";
+  parameter Modelica.SIunits.Time Ti(min=Modelica.Constants.small)=60
+    "Time constant of Integrator block" annotation (Dialog(enable=
+          controllerType == Modelica.Blocks.Types.SimpleController.PI or
+          controllerType == Modelica.Blocks.Types.SimpleController.PID));
+  parameter Modelica.SIunits.Time Td(min=0)=0.1
+    "Time constant of Derivative block" annotation (Dialog(enable=
+          controllerType == Modelica.Blocks.Types.SimpleController.PD or
+          controllerType == Modelica.Blocks.Types.SimpleController.PID));
+
+  Modelica.Blocks.Interfaces.RealInput masFloPum(
+    final unit="kg/s")
+    "Total mass flowrate of chilled water pumps"
+    annotation (Placement(transformation(extent={{-140,20},{-100,60}})));
+  Modelica.Blocks.Interfaces.RealInput dpMea(
+    final unit="Pa")
+    "Measured pressure difference"
+    annotation (Placement(transformation(extent={{-140,-60},{-100,-20}})));
+  Modelica.Blocks.Interfaces.RealOutput y[numPum](
+    unit="1",min=0,max=1) "Pump speed signal"
+    annotation (Placement(transformation(extent={{100,-10},{120,10}})));
+  Modelica.Blocks.Math.Product pumSpe[numPum] "Output pump speed"
+    annotation (Placement(transformation(extent={{40,-10},{60,10}})));
+  Buildings.Applications.DataCenters.ChillerCooled.Controls.VariableSpeedPumpStage pumStaCon(
```

Comment:
It may also need to propagate other parameters in the class, naming `criPoiFlo`, `deaBanFlo`, `criPoiSpe`, `deaBanSpe`.